### DISCUSSION TITLE
Issue calculating focal depth from cortical surface

### DISCUSSION THREAD

**Discussion by Jalyssa112** [simnibs/simnibs#315] [9. Results and analysis] [answered]
Hi, I am currently trying to find the distance from the cortical surface to a point in the brain to determine the focal depth, but am having trouble calculating the distance. Ideally I would like the distance from the greatest intensity on the surface to where the intensity starts to decrease in the brain. 

I have tried extracting the coordinates and e-field magnitudes in MATLAB given the code provided by simnibs but am not sure how to know what coordinates are the cortical surface and inside the brain.

If you have any suggestions that would be very helpful, thank you!

**Comment by TorgeW** [comment]
Can you please share the code that you used to run the simulation as well as the code you tried to use to get the coordinates?

**Comment by Jalyssa112** [comment]
Hello! I ran the simulations in SimNIBS and exported them to matlab with the code below. I also attached the code I used to get the coordinates 

```
x = load("simnibs_simulation_20240226-101326.mat")
head_mesh = mesh_load_gmsh4(...
    fullfile('C:\Users\jksmth10\Downloads\SI_Ped\SI_4.5-8.5\Mag_70', 'ped_TMS_1-0001_Magstim_70mm_Fig8_scalar.msh') ...
);

% Crop the mesh so we only have gray matter volume elements (tag 2 in the mesh)
gray_matter = mesh_extract_regions(head_mesh, 'region_idx', 2);
%mesh_show_surface(head_mesh)

% Electric fields are defined in the center of the elements
% get element centers
elm_centers = mesh_get_tetrahedron_centers(gray_matter);

%get fields at coordinates
field_coord = get_fields_at_coordinates(gray_matter, elm_centers)
```

**Reply by TorgeW** [reply]
so currently you are not doing anything with a surface. You are getting the field in the gray matter tetrahedron centers. 
The focal point would be the region with the strongest field, (use 99th percentile to avoid outliers). 

From there it depends on what exactly you want to do.

**Comment by Jalyssa112** [comment]
I removed the outliers based on the 99th percentile (I read SimNIBS uses 99.9th so I used that). I also combined the data so it would display the coordinates and electric field side by side [used a dummy variable, test].  I am hoping to determine the focal depth from the surface to a point in the gray matter. Is it possible to get that information from the data I have? 

```
%99.9th percentile
field_coord_perc = field_coord{1,2}.data;
field_coord_data = field_coord{1,1}.data;

perc = prctile(field_coord{1,2}.data,99.9,"all")
field_coord_perc(field_coord{1,2}.data >= perc) = []; 

%removed values in both coord and elec field 
    % [combined coord and e-field into one data matrix]
test = cat(2,field_coord{1,1}.data,field_coord{1,2}.data);
test2 = test(:,4) >= perc;
test(test2,:) = [];
```

**Reply by TorgeW** [reply]
This will give you the gray matter surface:
`gray_matter_surface = mesh_extract_regions(m, 'region_idx', 1002);`

You should be able to get the triangle centers using this:
https://github.com/simnibs/simnibs/blob/master/simnibs/matlab_tools/mesh_get_triangle_centers.m

This might also be interesting:
https://github.com/simnibs/simnibs/blob/master/simnibs/matlab_tools/mesh_get_fieldpeaks_and_focality.m

**Reply by Jalyssa112** [reply]
I am sorry for the slow reply, but that was very helpful thank you!

My last question is: I can use mesh_show_surface to show the surface of the gray or white matter, is it possible to visualize the volume (gray and white matter, corresponding to region_idx 1 and 2) in MATLAB. I have tried to add the elementary entities in SimNIBS and export the new file to MATLAB but the gmsh file will not load.

**Reply by TorgeW** [reply, marked as answer]
You have to write the msh file out and open it in gmsh.
The Matlab code cannot visualize volume.

**Reply by Jalyssa112** [reply]
Okay, thank you so much for the help!
